server/mvcc/backend/batch_tx.go

```diff
@@ -283,6 +283,10 @@ func (t *batchTxBuffered) CommitAndStop() {
 }
 
 func (t *batchTxBuffered) commit(stop bool) {
+	if t.backend.hooks != nil {
```

Comment:
Is this ever `nil`? Or should we just panic if `nil`?

Reply:
It's sometimes nil in context where we don't perform raft-applications: 
  - etcdctl (backup, snapshot, defrag, migrate) 
  - tests
  - verify (read only access)
In etcd server binary its not nil.